### ISSUE TITLE
Update text on the screen when updated state

### ISSUE THREAD

**Issue by HulioEglesias** [flutter/flutter#71317]
Why my Statefull widget don't update?
I have a 2 screens List of items and Describe of one item.
ListScreen has a list of avatars and titles. ItemScreen has a avatars and titles from tappable listTile(through `Navigator.pushNamed(context, 'route, arguments: { item: this.item })` and describe.
When I navigate to DescribeScreen, I show avatar and title and start request for describe. But when describe is loaded I can't see that(i can see only after update screen by hot reload).
Before `item.updateIngredients();` `item.ingredients == null` and after it's a List.

ItemScreen
```
class _ItemScreenState extends State<ItemScreen> {
  CosmeticItem item;

  @override
  Widget build(BuildContext context) {
    final Map arguments = ModalRoute.of(context).settings.arguments as Map;
    item = arguments['item'];

    setState(() {
      item.updateIngredients();
    });

    return Scaffold(
      body: Column(
        children: [
          Hero(
            tag: 'hero_image_${item.id}',
            child: Image(
              image: CachedNetworkImageProvider(item.imageSrc),
              fit: BoxFit.fitWidth,
              width: MediaQuery.of(context).size.width,
              height: 300.0,
            ),
          ),
          Padding(
            padding: EdgeInsets.all(18.0),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(
                  item.brand,
                  style: TextStyle(
                    fontSize: 18.0,
                  ),
                ),
                Text(
                  item.name,
                  style: TextStyle(
                    fontSize: 18.0,
                    fontWeight: FontWeight.bold,
                  ),
                ),
                Text(
                  item.description,
                  style: TextStyle(
                    fontSize: 14.0,
                  ),
                ),
                SingleChildScrollView(
                  child: (item.ingredients != null)
                      ? _IngredientList(item.ingredients)
                      : CircularProgressIndicator(),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}

class _IngredientList extends StatelessWidget {
  _IngredientList(this.ingredients);
  final List<CosmeticIngredient> ingredients;

  @override
  Widget build(BuildContext context) {
    return Column(
      children: (ingredients.length < 1)
          ? []
          : ingredients.map((ing) => Text(ing.name)).toList(),
    );
  }
}

```
Item class
```
import 'dart:convert';
import 'package:http/http.dart' as http;

class CosmeticItem {
  CosmeticItem({this.id, this.name, this.barcode, this.imageSrc, this.brand, this.description, this.ingredients});
  int id;
  String imageSrc;
  String name;
  String barcode;
  String brand;
  String description;
  List<CosmeticIngredient> ingredients;

  void updateIngredients() async {
    http.Response resp = await http
        .get('http://domain.com/api/v1/admin/products/$id');
    var body = jsonDecode(resp.body);
    ingredients = [];
    for(var ingr in body['ingredients']) {
      ingredients.add(CosmeticIngredient(
        id: ingr['id'],
        name: ingr['name'],
        description: ingr['description'],
        herbal: ingr['herbal'],
        additional: ingr['additional'],
        active: ingr['active']
      ));
    }
  }
}
class CosmeticIngredient {
  CosmeticIngredient({this.id, this.name, this.description, this.herbal, this.additional, this.active});
  int id;
  String name;
  String description;
  bool herbal;
  bool additional;
  bool active;
}
```

**Comment by iapicca** [comment]
@HulioEglesias 
this seems more a request for help to solve a problem with your code than a bug with flutter itself
I recommend to ask a question on stackoverflow 
as indicated by the choice `I want help writing my application` in the [new issue selection](https://github.com/flutter/flutter/issues/new/choose)

**Comment by darshankawar** [comment]
@HulioEglesias 
As suggested above, please follow the link above to get help on your issue. Also see https://flutter.dev/community for resources and if you need help with your code, please see https://www.reddit.com/r/flutterhelp/
Closing, as this isn't an issue with Flutter itself. If you disagree, please write in the comments and I will reopen it.
Thank you
